core/src/main/java/io/atomix/core/lock/impl/DistributedLockService.java

```diff
@@ -160,7 +160,7 @@ protected void unlock(Commit<Unlock> commit) {
   }
 
   private void releaseSession(Session session) {
-    if (lock.session == session.sessionId().id()) {
+    if (lock != null && lock.session == session.sessionId().id()) {
```

Comment:
@kuujo shouldn't this be lock == null || ..

Reply:
As to drain the queue.

Reply:
If `lock` is null then the queue should be empty